packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +128,64 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot ??= (async () => {
+				this.latestSnapshot = await getLatestSnapshotInfo(
+					this.mc,
+					this.storageAdapter,
+					supportGetSnapshotApi,
+				);
+				this.updateSnapshotAndProcessedOpsMaybe();
+			})();
+
 			return { baseSnapshot, version: undefined };
 		}
 	}
 
+	/**
+	 * Updates class snapshot and processedOps if we have a new snapshot and it's among processedOps range.
+	 */
+	private updateSnapshotAndProcessedOpsMaybe() {
+		if (this.latestSnapshot === undefined) {
+			return;
+		}
+		this.snapshotRefreshed?.();
```

Comment:
shouldn't this be called around 185 such that its called when the snapshot is actually refreshed?

Reply:
It depends on what we want tests to check and when. I was more interested in knowing when the new snapshot was fetched than when we do the this.snapshot update. That allowed me to track the catch up case a bit more clearly. I'll reconsider where I put this and how I call it after updating some tests